src/pages/Intents/ResponseForm.js

```diff
@@ -0,0 +1,42 @@
+import PropTypes from 'prop-types'
+import React, { useState, useEffect } from 'react'
+import { TextField } from '../../components/field/TextField'
+
+const ResponseForm = ({ data = {}, onChange }) => {
+    const [responseData, setResponseData] = useState(data)
+
+    useEffect(() => {
+        setResponseData(data)
+    }, [data])
+
+    const handleChange = (field, value) => {
+        const updated = { ...responseData, [field]: value }
+        setResponseData(updated)
+        onChange('response.data', updated)
```

Comment:
why not just fully make use of the setByPath function you added and update only the specific field e.g update onChange('response.data.argument',value), rather than updating the whole data object and then updating that?